pygount/analysis.py

```diff
@@ -521,6 +551,47 @@ def name_regexps_to_skip(self):
     def name_regexps_to_skip(self, regexps_or_pattern_text):
         self._name_regexp_to_skip = pygount.common.regexes_from(regexps_or_pattern_text, self.name_regexps_to_skip)
 
+    @staticmethod
+    def is_valid_git_link(git_link: str) -> bool:
+        # Regex to check valid  GIT Repository
+        # from https://stackoverflow.com/questions/2514859/regular-expression-for-git-repository
+        git_regex = re.compile(r"((git|ssh|http(s)?)|(git@[\w\.-]+))(:(\/\/)?)([\w\.@\:\/\-~]+)(\.git)(\/)?")
+
+        if git_regex.match(git_link):
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def repo_arguments(git_link: str):
+        """
+        Returns the git link that ends with .git and branch name if it specified at the end with @branch
+        :param git_link: input of the git link that ends with .git or .git@branch
+        """
+        # example: master, feature-branch
+        # at this state it's set to '@master' or ''
+        git_branch = git_link.split(".git")[-1]
+
+        # example: https://github.com/roskakori/pygount.git
+        git_link = git_link.split(".git")[0] + ".git"
+
+        # set the argument for the clone command if any branch is specified after `.git/`
+        if git_branch != "":
+            # set from '@master' to 'master'
+            git_branch = "--branch " + git_branch.split("@")[1]
+
+        return git_link, git_branch
+
+    def _set_temp_path_from_git_link(self, source_patterns):
```

Comment:
Confusing name. Ideally functions are either named after what they do (if the change things) or what the return (if the just compute a result).

Reply:
Changed in [86cbce5](https://github.com/roskakori/pygount/pull/111/commits/86cbce5e2f49caf80242b97c3aa5c0708b4abf03)

Reply:
Changed to GitStorage